src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkWatchDog.java

```diff
@@ -32,19 +32,28 @@
 public class ZkWatchDog extends ZkClient implements DataCallback {
 
     private static final String SIGNAL = "signal";
+    private static final String BUILD_VERSION = "build-version";
+
     @VisibleForTesting
     protected String signalPath;
     private Signal signal;
 
+    @VisibleForTesting
+    protected String buildVersionPath;
+    private String buildVersion;
+
     @VisibleForTesting
     Set<LifeCycleObserver> observers = new HashSet<>();
+    @VisibleForTesting
+    Set<BuildVersionObserver> buildVersionObservers = new HashSet<>();
```

Comment:
As for me, it looks like you need some "generic" (with java generics) class responsible for publishing to a single node (with shared zk-client if it is possible) and some closure for the set of such publishers. This approach is much more scalable than the current one.